### ISSUE TITLE
Unable to reproduce MoleculeNet results using MPNN

### ISSUE THREAD

**Issue by ujvl** [deepchem/deepchem#2344]
## 🐛 Bug

I'm trying to repro the table 3 results from the [MoleculeNet paper](https://arxiv.org/pdf/1703.00564.pdf). MPNN trained on ESOL should yield a test RMSE of 0.58, but I get ~0.65 with the provided example [delaney_MPNN.py](https://github.com/deepchem/deepchem/blob/master/examples/delaney/delaney_MPNN.py). I also tried to use the hyperparameters listed in the repo [here](https://github.com/deepchem/deepchem/blob/master/deepchem/molnet/preset_hyper_parameters.py#L176) but that doesn't seem to produce the expected performance either.

Are either of these the correct hparam configurations to use, and if not, where can I find the ones used to produce the results in the paper? Alternatively, is there anything else I'm missing in training the model?

## To Reproduce

Run [delaney_MPNN.py](https://github.com/deepchem/deepchem/blob/master/examples/delaney/delaney_MPNN.py) (with `test_scores = model.evaluate(test_dataset, [metric], transformers)` appended to the end and metric replaced w/ `dc.metrics.Metric(dc.metrics.rms_score, np.mean)`).

## Expected behavior

Test RMSE of 0.58 for MPNN x ESOL.

## Environment

* OS: Ubuntu
* Python version: 3.6
* DeepChem version: 2.4.0
* RDKit version (optional): 2020.09.1
* TensorFlow version (optional): 2.3.0

**Comment by peastman** [comment]
I can make an educated guess.  I believe the original MoleculeNet paper used random splitting for that dataset, which overestimates how well models will do on independent data.  The current DeepChem code uses scaffold splitting by default, which is somewhat better.  Try adding the argument `splitter='random'` to the `load_delaney()` call.  Does that reproduce the original number?

**Comment by ujvl** [comment]
I've tried both 'index' and 'random' with no luck. Not sure if this might point to a larger problem, but I don't get deterministic results when I run the script multiple times (fixing the seed, hyperparameters, and split to index). Is that to be expected?

I also just found other (different) hparam configurations saved in the tar linked here: https://github.com/deepchem/deepchem/blob/678c621498df1154782ffca885965363fcb0a0ae/examples/benchmark.py#L30
Wasn't able to reproduce the results with these ones either but do you know if these are the ones I should be using? They seem to be different from those mentioned elsewhere in this repo.
